mesonbuild/interpreter.py

```diff
@@ -2969,21 +3003,59 @@ def _handle_featurenew_dependencies(self, name):
         elif name == 'openmp':
             FeatureNew('OpenMP Dependency', '0.46.0').use(self.subproject)
 
+    def notfound_dependency(self):
+        return DependencyHolder(NotFoundDependency(self.environment), self.subproject)
+
     @FeatureNewKwargs('dependency', '0.49.0', ['disabler'])
     @FeatureNewKwargs('dependency', '0.40.0', ['method'])
     @FeatureNewKwargs('dependency', '0.38.0', ['default_options'])
     @disablerIfNotFound
     @permittedKwargs(permitted_kwargs['dependency'])
     def func_dependency(self, node, args, kwargs):
-        self.validate_arguments(args, 1, [str])
-        name = args[0]
+        fallbacks = listify(args)
+        if len(fallbacks) == 0 or not isinstance(fallbacks[0], str):
+            raise InvalidArguments('dependency first positional argument must be a string')
+        if not isinstance(args[0], str):
+            if 'fallback' in kwargs:
+                raise InvalidArguments('dependency must not have fallback keyword argument '
+                                       'when fallbacks are provided as positional arguments')
+            FeatureNew('dependency fallbacks', '0.50.0').user(self.subproject)
+        name = fallbacks[0]
         display_name = name if name else '(anonymous)'
 
         disabled, required, feature = extract_required_kwarg(kwargs, self.subproject)
         if disabled:
             mlog.log('Dependency', mlog.bold(display_name), 'skipped: feature', mlog.bold(feature), 'disabled')
-            return DependencyHolder(NotFoundDependency(self.environment), self.subproject)
+            return self.notfound_dependency()
+
+        # Fallback only to subprojects in forcefallback mode, unless there is no
+        # subproject in the list.
+        if self.coredata.get_builtin_option('wrap_mode') == WrapMode.forcefallback:
+          subproject_fallbacks = [item for item in fallbacks if isinstance(item, dict) and 'subproject' in item]
```

Comment:
__[Flake8]__ <sub>posted by Sider</sub>

[E111] indentation is not a multiple of four